exploration/bidi-usability.md

```diff
@@ -348,47 +407,50 @@ the results or debug what is wrong with their messages.
 By contrast, if users insert too many or the wrong controls using the recommended design,
 the _message_ would still be functional and would emit no undesired characters.
 
+### Super-loose isolation
 
-### Loose isolation
+Add isolates and strongly directional marks to required and optional whitespace in the syntax.
+This would permit users to get the effects described by the above design,
+as long as they use isolates/marks in a "responsible" way.
 
-Apply bidi isolates in a slightly different way.
-The main differences to the proposed solution are:
-1. The open/close isolate characters are not syntactically required to be paired.
-   This avoids introducing parse errors for missing or required invisible characters,
-   which would lead to bad user experiences.
-2. Rather than patching the `name` rule with an optional trailing LRM/RLM/ALM,
-   allow for its proper isolation.
+(Omitting other changes found in #673)
 
-Quoted patterns, quoted literals, and names may be isolated by LRI/RLI/FSI...PDI.
-For names and quoted literals, the isolate characters are outside the body of the token,
-but for quoted patterns, the isolates are in the middle of the `{{` and `}}` characters.
-This avoids adding a lookahead requirement for detecting a `complex-message` start,
-and differentiates a `quoted-pattern` from a `quoted` `key` in a `variant`.
+```abnf
+; strongly directional marks and bidi isolates
+; ALM / LRM / RLM / LRI / RLI / FSI / PDI
+bidi = %x061C / %x200E / %x200F / %x2066-2069
 
-Expressions and markup may be isolated by LRI...PDI immediately within the `{` and `}`.
+; optional whitespace
+owsp = *( s / bidi )
 
-An LRI is allowed immediately after a newline outside patterns and within expressions.
-This is intended to allow left-to-right representation for "code"
-even if it contains a newline followed by content
-that could otherwise prompt the paragraph direction to be detected as right-to-left.
+; required whitespace
+wsp = [ owsp ] 1*s [ owsp ]
 
-```abnf
-name           = [open-isolate] name-start *name-char [close-isolate]
-quoted         = [open-isolate] "|" *(quoted-char / quoted-escape) "|" [close-isolate]
-quoted-pattern = "{" [open-isolate] "{" pattern "}" [close-isolate] "}"
+; whitespace characters
+s = ( SP / HTAB / CR / LF / %x3000 )
+```
+
+**Pros**
+- Avoids problems with syntax errors that users and tools might find difficult to debug.
+- Effective if used carefully.
+- Addresses need to comply with UAX#31
 
-literal-expression    = "{" [LRI] [s] literal [s annotation] *(s attribute) [s] [close-isolate] "}"
-variable-expression   = "{" [LRI] [s] variable [s annotation] *(s attribute) [s] [close-isolate] "}"
-annotation-expression = "{" [LRI] [s] annotation *(s attribute) [s] [close-isolate] "}"
+**Cons**
+- Can be used irresponsibly, including enabling some Trojan Source cases (UAX#55)
 
```

Comment:
```suggestion
    - However, linters could support "responsible" use of bidi characters. (It is likely that many implementations will end up needing to support linting, such as to verify the right number and datatype of input parameters, etc.
```

Reply:
This would be better as a pro perhaps? "Irresponsibly" might be the wrong word in the 'con', although it smacks of code attacks, which is the idea. This design enables the right thing by also enabling many wrong things. On the other hand, a lot of the time we'll be lucky if tools implement anything or users bother to insert the isolates/marks.

Reply:
See most recent edits, which incorporate your suggestion into the 'con' text as well as adding more prose to the design option such that this is (hopefully) clear. I also added one more design, which tries to split-the-difference